hook-common/src/kafka_messages/app_metrics.rs

```diff
@@ -77,19 +88,88 @@ where
     serializer.serialize_str(category_str)
 }
 
-fn serialize_error_type<S>(error_type: &ErrorType, serializer: S) -> Result<S::Ok, S::Error>
+fn deserialize_category<'de, D>(deserializer: D) -> Result<AppMetricCategory, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: String = Deserialize::deserialize(deserializer)?;
+
+    let category = match &s[..] {
+        "processEvent" => AppMetricCategory::ProcessEvent,
+        "onEvent" => AppMetricCategory::OnEvent,
+        "scheduledTask" => AppMetricCategory::ScheduledTask,
+        "webhook" => AppMetricCategory::Webhook,
+        "composeWebhook" => AppMetricCategory::ComposeWebhook,
+        _ => {
+            return Err(serde::de::Error::unknown_variant(
+                &s,
+                &[
+                    "processEvent",
+                    "onEvent",
+                    "scheduledTask",
+                    "webhook",
+                    "composeWebhook",
+                ],
+            ))
+        }
+    };
+
+    Ok(category)
+}
+
+fn serialize_error_type<S>(error_type: &Option<ErrorType>, serializer: S) -> Result<S::Ok, S::Error>
 where
     S: Serializer,
 {
     let error_type = match error_type {
-        ErrorType::Connection => "Connection Error".to_owned(),
-        ErrorType::Timeout => "Timeout".to_owned(),
-        ErrorType::HttpStatus(s) => format!("HTTP Status: {}", s),
-        ErrorType::Parse => "Parse Error".to_owned(),
+        Some(error_type) => error_type,
+        None => return serializer.serialize_none(),
+    };
+
+    let error_type = match error_type {
+        ErrorType::ConnectionError => "Connection Error".to_owned(),
+        ErrorType::TimeoutError => "Timeout".to_owned(),
+        ErrorType::BadHttpStatus(s) => format!("HTTP Status: {}", s),
```

Comment:
I don't know if these names matter too much, but for consistency:

```suggestion
        ErrorType::TimeoutError => "Timeout Error".to_owned(),
        ErrorType::BadHttpStatus(s) => format!("Bad HTTP Status: {}", s),
```